### ISSUE TITLE
Problems raised about missing argument when calling a function decorated by celery.shared_task

### ISSUE THREAD

**Issue by remihuguet** [microsoft/pylance-release#39]
## Environment data

-   Language Server version: 2020.6.1
-   OS and version: Ubuntu 20.04
-   Python version 3.7.7, virtual env with requirements joined
[requirements_pypi.txt](https://github.com/microsoft/pylance-release/files/4868771/requirements_pypi.txt)

## Expected behaviour

When calling a task defined with `celery.app.shared_task` decorator, and naming argument, Pylance raises a problem "Argument missing parameter for 'fun'. See code snippet.

![problems](https://user-images.githubusercontent.com/19345704/86452389-879a6d00-bd1c-11ea-91a8-ca8116239c05.png)

## Actual behaviour

No problems should be raised.

## Logs

```
[FG] parsing: tests.py (10ms)
[FG] binding: tests.py (2ms)
[BG] analyzing: tests.py ...
[BG]   parsing: tests.py (7ms)
[BG]   binding: tests.py (1ms)
[BG]   checking: tests.py (16ms)
[BG] analyzing: tests.py (25ms)
[FG] parsing: tests.py (5ms)
[FG] binding: tests.py (2ms)
[BG] analyzing: tests.py ...
[BG]   parsing: tests.py (1ms)
[BG]   binding: tests.py (0ms)
[BG]   checking: tests.py (29ms)
[BG] analyzing: tests.py (30ms)
```

## Code Snippet / Additional information

```python
from celery.app import shared_task

@shared_task
def function_one(argument):
    return []

def function_two():
    return function_one(argument=2)
```

**Comment by erictraut** [comment]
Pylance requires type stubs to do the best job possibly with type analysis. In this case, I'm presuming that you don't have a type stub available for celery, so it is doing its best to infer types from the library implementation based on the return type(s) of `shared_task`. In this particular case, `shared_task` returns two different types, and the type checker can't tell which one is returned for this particular call site. Both types are functions, but one of them is a function that requires a parameter called `fun`, hence the error.

The best workaround in this case is to find (or create) a type stub for the celery package. A type stub will provide the proper annotations so the type checker can perform analysis correctly in this case without emitting an error.

If you can't find a type stub and don't want to create one, you can suppress this error using "# type: ignore".

**Comment by erictraut** [comment]
I've updated the "missing parameter" diagnostic rule to be off by default when typeCheckingMode is "off". Assuming that you haven't set typeCheckingMode to "basic" or "strict", this diagnostic will be suppressed. This change will be in the next version of Pylance.

**Comment by remihuguet** [comment]
ok thanks you very much for your feedback. I'll try to add so type stub for celery :+1: 

By the  way, this is a great extension, very nice work :clap:

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.7.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202070-9-july-2020
